WalletWasabi.Tests/UnitTests/Tor/Socks5/TorHttpPoolTests.cs

```diff
@@ -18,33 +21,98 @@ namespace WalletWasabi.Tests.UnitTests.Tor.Socks5
 	public class TorHttpPoolTests
 	{
 		/// <summary>
-		/// Tests <see cref="TorHttpPool.SendAsync(HttpRequestMessage, bool, CancellationToken)"/> method.
+		/// Tests that <see cref="TorHttpPool.SendAsync(HttpRequestMessage, IIdentity, CancellationToken)"/> method respects provided identity.
 		/// </summary>
-		/// <seealso href="https://stackoverflow.com/questions/9114053/sample-on-namedpipeserverstream-vs-namedpipeserverclient-having-pipedirection-in"/>
 		[Fact]
-		public async Task SendAsync()
+		public async Task UseCorrectIdentitiesAsync()
+		{
+			Common.GetWorkDir();
+			Logger.SetMinimumLevel(LogLevel.Trace);
+
+			using CancellationTokenSource timeoutCts = new(TimeSpan.FromMinutes(1));
+
+			IIdentity defaultIdentity = DefaultIdentity.Instance;
+			IIdentity aliceIdentity = new RandomIdentity(canTorCircuitBeReused: true); // Random but a non-changing identity.
+			IIdentity bobIdentity = new RandomIdentity(canTorCircuitBeReused: true); // Random but a non-changing identity.			
```

Comment:
Bob posting the outputs with different identities every time. Maybe it is not important in tests.

Reply:
To limit scope of this PR, I just want to add a new feature and not change existing code too much unless there is a strong reason. It can be done in a follow-up but I'm not an expert on CJ code so I don't change it needlessly.

We can certainly file an issue for this or you can add a commit under your name.

Reply:
It does not matter in this test. 

By the way currently we are creating new Bobs for every single output. https://github.com/zkSNACKs/WalletWasabi/blob/b2ab91831a1f59b0a265f0ff3ca2de713cc65a92/WalletWasabi/CoinJoin/Client/Clients/CoinJoinClient.cs#L429

Maybe WabiSabi will have only one Bob with NewIdentityPerRequest mode.